scripts/linux-prereqs.sh

```diff
@@ -51,90 +47,117 @@ sudoif()
     "$@"
 }
 
+# Utility function that waits for any existing installation operations to complete
+# on Debian/Ubuntu based distributions and then calls apt-get
+aptsudoif() 
+{
+    while sudoif fuser /var/lib/dpkg/lock >/dev/null 2>&1; do
+        echo -ne "Waiting for other package operations to complete.\r"
+        sleep 0.2
+        echo -ne "Waiting for other package operations to complete..\r"
+        sleep 0.2
+        echo -ne "Waiting for other package operations to complete...\r"
+        sleep 0.2
+        echo -ne "\r\033[K"
+    done
+    sudoif apt-get "$@"
+}
+
+# If not already root, validate user has sudo access and error if not.
+if [ $(id -u) -ne 0 ]; then
+
+# Can't indent or text will be indented
+cat << EOF
+To begin the installation process, your OS will now ask you to enter your
+admin / root (sudo) password.
+
+EOF
+    # Validate user actually can use sudo
+    sudo -v > /dev/null 2>&1
+    if [ $? -ne 0 ]; then
+
+# Can't indent or text will be indented
+cat << EOF
+(!) Dependency installation failed! You do not have the needed admin / root
+    access to install Live Share's dependencies. Contact your system admin
+    and ask them to install the required libraries described here:
+    https://aka.ms/vsls-docs/linux-required-lib-details
+EOF
+        exitscript 3
+    else
+        echo ""
+    fi
+fi
+
 #openSUSE - Has to be first as apt is aliased to zypper
 if type zypper > /dev/null 2>&1; then
```

Comment:
t we are not detecting the distribution but we are assuming it based on the package manager, this may cause an issue if the script is running on a distro that uses a package manager we have listed but has different packages in the repositories. Should we change approach? I suggest to parse `/etc/os-release`

Reply:
That is not that reliable either from what we have seen. SuSE is the only one that does this from what I have seen.

Reply:
For example, apt-get works for any distribution down stream of Debian which is a massive set some of which alter this value. The package manager is the most reliable.

Reply:
Oh ok so the plan is supporting derivate distros too! Seems good to me